scipy/signal/_splinemodule.cc

```diff
@@ -201,49 +210,56 @@ static PyObject *IIRsymorder1_ic(PyObject *NPY_UNUSED(dummy), PyObject *args)
   const npy_intp sz[2] = {M, 1};
   dtype = PyArray_DescrFromType(thetype);
   out = (PyArrayObject *)PyArray_Empty(2, sz, dtype, 0);
-  if (out == NULL) goto fail;
+  if (out == NULL) {
+    // Duplicate the `goto fail` block to avoid "crosses initialization" build errors.
+    Py_XDECREF(a_sig);
+    Py_XDECREF(out);
+    return NULL;
+  }
 
   switch (thetype) {
   case NPY_FLOAT:
     {
       float rz1 = z1.real;
 
       if ((precision <= 0.0) || (precision > 1.0)) precision = 1e-6;
-      ret = S_SYM_IIR1_initial(rz1, (float *)PyArray_DATA(a_sig),
-                               (float *)PyArray_DATA(out), M, N,
-                               (float )precision);
+        ret = _sym_iir1_initial(rz1, static_cast<float *>(PyArray_DATA(a_sig)),
+                               static_cast<float *>(PyArray_DATA(out)), M, N,
+                               static_cast<float>(precision));
     }
     break;
   case NPY_DOUBLE:
     {
       double rz1 = z1.real;
 
       if ((precision <= 0.0) || (precision > 1.0)) precision = 1e-11;
-      ret = D_SYM_IIR1_initial(rz1, (double *)PyArray_DATA(a_sig),
-                               (double *)PyArray_DATA(out), M, N,
-                               precision);
+      ret = _sym_iir1_initial(rz1, static_cast<double *>(PyArray_DATA(a_sig)),
+                              static_cast<double *>(PyArray_DATA(out)), M, N,
+                              precision);
```

Comment:
It's _slightly_ less of an issue if the indentation matches the `if`, but I'd still unwrap it in the same way as noted above, and so on for every single-line precision-condition in this file

(162ed2d602240bb291b0a927ea7f94426789b36b; though changing all those single-line `if`'s would actually make a pretty good stand-alone commit)

Reply:
I'd much rather minimize manual reformatting and re-add a clang-format commit. WDYT?

Reply:
Completely fine to add a separate commit doing clang-format at the end. Only in case that clang-format _does not_ break up or separate these single-line if's, I still think we should fix them up manually.